module/VuFind/tests/unit-tests/src/VuFindTest/I18n/SorterTest.php

```diff
@@ -0,0 +1,353 @@
+                ],
+                0
+            ],
+            [
+                [
+                    'strings' => ['b', 'a'],
+                    'locale' => 'en',
+                    'respectLocale' => true,
+                ],
+                1
+            ],
+            [
+                [
+                    'strings' => ['a', 'A'],
+                    'locale' => 'en',
+                    'respectLocale' => true,
+                ],
+                -1
+            ],
+            [
+                [
+                    'strings' => ['č', 'd'],
+                    'locale' => 'cs',
+                    'respectLocale' => false,
+                ],
+                1
+            ],
+            [
+                [
+                    'strings' => ['č', 'd'],
+                    'locale' => 'cs',
+                    'respectLocale' => true,
+                ],
+                -1
+            ],
+        ];
+    }
+
+    /**
+     * Test compare function
+     *
+     * @dataProvider compareProvider
+     *
+     * @return void
+     */
+    public function testCompare($test, $expected)
+    {
+        $sorter = $this->createSorter($test['locale'], $test['respectLocale']);
+        $result =  $sorter->compare($test['strings'][0], $test['strings'][1]);
+        if ($expected === 1) {
+            $this->assertGreaterThanOrEqual($expected, $result);
+        } elseif ($expected === -1) {
+            $this->assertLessThanOrEqual($expected, $result);
+        } else {
+            $this->assertEquals($expected, $result);
+        }
+    }
+
+    /**
+     * Data Provider for sort tests
+     *
+     * @return array
+     */
+    public static function sortProvider(): array
+    {
+        return [
+            [
+                [
+                    'input' => ['a', 'c', 'b'],
+                    'locale' => 'en',
+                    'respectLocale' => false,
+                ],
+                ['a', 'b', 'c'],
+            ],
+            [
+                [
+                    'input' => ['a', 'c', 'b'],
+                    'locale' => 'en',
+                    'respectLocale' => true,
+                ],
+                ['a', 'b', 'c'],
+            ],
+            [
+                [
+                    'input' => ['a', 'č', 'd', 'c'],
+                    'locale' => 'cs',
+                    'respectLocale' => false,
+                ],
+                ['a', 'c', 'd', 'č'],
+            ],
+            [
+                [
+                    'input' => ['a', 'č', 'd', 'c'],
+                    'locale' => 'cs',
+                    'respectLocale' => true,
+                ],
+                ['a', 'c', 'č', 'd'],
+            ],
+            [
+                [
+                    'input' => ['100', '3', '10', '2', '1'],
+                    'locale' => 'cs',
+                    'respectLocale' => true,
+                ],
+                ['1', '2', '3', '10', '100'],
```

Comment:
I'm a little surprised that we're getting a natural sort here -- I would have expected 10 and 100 to come between 1 and 2. Is this because the strings are getting treated as integers behind the scenes? If so, is that a bug or a feature? Either way, maybe it would be good to also have a test case similar to the one in your natcase tests below (e.g. a1, a2, a10), just to show differentiated behavior.